src/tests/conftest.py

```diff
@@ -0,0 +1,160 @@
+# *******************************************************
+# Copyright (c) VMware, Inc. 2020-2023. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""Flask mocking app"""
+
+import threading
+import socket
+# import ssl
+import traceback
+import json
+import pprint
+import logging
+import pytest
+
+from flask import Flask, request, jsonify
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+app = Flask(__name__)
+
+# cert_file = '../cert.pem'
+# key_file = '../cert.pem'
+tcp_tls_server_port = 8888
+tcp_server_port = 8887
+udp_server_port = 8886
+
+
+def pytest_configure():
+    """Pytest Global Variables"""
+    pytest.alert_search_request = None
+    pytest.alert_search_response = None
+
+#
+# Carbon Black Cloud Mocked Endpoints
+#
+
+
+@app.route('/appservices/v6/orgs/<org_key>/alerts/_search', methods=['POST'])
+def alert_search(org_key):
+    """alert_search"""
+    logger.info("Fetched Alerts")
+
+    # Save the request for verification
+    pytest.alert_search_request = request.get_json()
+
+    output = {
+        "num_found": 0,
+        "num_available": 0,
+        "results": []
+    }
+    if callable(pytest.alert_search_response):
+        output = pytest.alert_search_response(pytest.alert_search_request)
+    else:
+        output = pytest.alert_search_response
+
+    return jsonify(output)
+
+
+#
+# Syslog Output Mocked Servers
+#
+
+
+@app.route('/http_out', methods=['POST'])
+def http_out():
+    """http_out"""
+    try:
+        content = request.json
+        logger.info(content)
+    except Exception:
+        logger.info(traceback.format_exc())
+    return jsonify({})
+
+
+def udp_server_func():
+    """udp_server"""
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    server_address = ('0.0.0.0', udp_server_port)
+    print("udp_server is listening on port {}".format(udp_server_port))
+    sock.bind(server_address)
+
+    while True:
+        data, address = sock.recvfrom(4096)
+        print(address)
+        print(len(data))
+        print(repr(data))
+
+
+def tcp_server_func():
+    """tcp_server"""
+    # Create a TCP/IP socket
+    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
Would it make sense to set socket option SO_REUSEADDR to avoid bind() errors when restarting the server (e.g due to some config change)?

Reply:
I don't have anything consuming this yet as it's a future story. This code was copied from the previous test server. Will adjust to make the code better then